play_book_pdf_tool/play_book_pdf_tool.py

```diff
@@ -89,8 +93,13 @@ def add_metadata(base_path, pdf):
     language = manifest['language']
     title, authors, pub_date, num_pages, publisher = itemgetter('title', 'authors', 'pub_date', 'num_pages',
                                                                 'publisher')(manifest['metadata'])
-
-    year, month, day = [int(x) for x in pub_date.split('.')]
+    pub_date_parts = pub_date.split('.')
+    year = int(pub_date_parts[0])
+    month = int(pub_date_parts[1]) if len(pub_date_parts) > 1 else 1
+    day = int(pub_date_parts[2]) if len(pub_date_parts) > 2 else 1
```

Comment:
XMP dates support partial dates that omit the day, or omit the day and month: https://developer.adobe.com/xmp/docs/XMPNamespaces/XMPDataTypes/#date

Could the `xmp:CreateDate` property be updated accordingly instead of faking `1`s when they are missing?

Reply:
I didn't realize there was support for not full date, I was at that point just avoiding crashing; I'll have to look

Reply:
I didn't realize there was support for not full date, I was at that point just avoiding crashing)

But knowing this, your proposal looks like the best solution)

Reply:
https://github.com/devnoname120/google-play-book-downloader/pull/14